cf-service-count.py

```diff
@@ -0,0 +1,138 @@
+            cache[cf_svc_call_str] = cf_svc_url_data
+            service_map["service_offering"].append(cache[cf_svc_call_str]["entity"]["label"])
+
+        logging.debug("fetching service plan info")
+        cf_svc_plan_call = ['cf', 'curl', si["entity"]["service_plan_url"]]
+        cf_svc_plan_call_str = "".join(cf_svc_plan_call)
+
+        if cf_svc_plan_call_str in cache:
+            cache_hits += 1
+            logging.debug("cache hit on {0}".format(cf_svc_plan_call))
+            service_map["service_plan"].append(cache[cf_svc_plan_call_str]["entity"]["name"])
+        else:
+            cache_misses += 1
+            logging.debug("cache miss on {0}".format(cf_svc_plan_call))
+            cache[cf_svc_plan_call_str] = None
+            cf_svc_plan_out = subprocess.run(cf_svc_plan_call, shell=False, stdout=subprocess.PIPE)
+            cache[cf_svc_plan_call_str] = json.loads(cf_svc_plan_out.stdout)
+            service_map["service_plan"].append(cache[cf_svc_plan_call_str]["entity"]["name"])
+
+        logging.debug("fetching space info")
+        cf_space_call = ["cf", "curl", "/v3/spaces/{0}".format(si["entity"]["space_guid"])]
+        cf_space_call_str = "".join(cf_space_call)
+
+        if cf_space_call_str in cache:
+            cache_hits += 1
+            logging.debug("cache hit on {0}".format(cf_space_call))
+            service_map["space"].append(cache[cf_space_call_str]["name"])
+        else:
+            cache_misses += 1
+            logging.debug("cache miss on {0}".format(cf_space_call))
+            cf_space_out = subprocess.run(cf_space_call, shell=False, stdout=subprocess.PIPE)
+            cache[cf_space_call_str] = json.loads(cf_space_out.stdout)
+            service_map["space"].append(cache[cf_space_call_str]["name"])
+
+        logging.debug("fetching org info")
+        # it's a full URL, and we need to split on it.
+        # it's also guaranteed to be in the cache because of the previous call.
+        org_link = "/" + "/".join(cache[cf_space_call_str]["links"]["organization"]["href"].split("/")[3:])
+        cf_org_call = ['cf', 'curl', org_link]
+        cf_org_call_str = "".join(cf_org_call)
+
+        if cf_org_call_str in cache:
+            cache_hits += 1
+            logging.debug("cache hit on {0}".format(cf_org_call))
+            service_map["org"].append(cache[cf_org_call_str]["name"])
+        else:
+            cache_misses += 1
+            logging.debug("cache miss on {0}".format(cf_org_call))
+            cf_org_out = subprocess.run(cf_org_call, shell=False, stdout=subprocess.PIPE)
+            cache[cf_org_call_str] = json.loads(cf_org_out.stdout)
+            service_map["org"].append(cache[cf_org_call_str]["name"])
+
+    # this is just because it's easiest for the conversion to csv.
+    df = pd.DataFrame(service_map)
+    if args.file_name != "":
+        df.to_csv(args.file_name, index=False)
+    else:
+        df.to_csv("service-count.csv", index=False)
```

Comment:
how about stdout as a default instead?

Reply:
It would be interspersed with the log messages so it wouldn't be helpful.